community/modules/scheduler/htcondor-central-manager/variables.tf

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "Project in which HTCondor pool will be created"
+  type        = string
+}
+
+variable "deployment_name" {
+  description = "HPC Toolkit deployment name. HTCondor cloud resource names will include this value."
+  type        = string
+}
+
+variable "labels" {
+  description = "Labels to add to resources. List key, value pairs."
+  type        = map(string)
+}
+
+variable "region" {
+  description = "Default region for creating resources"
+  type        = string
+}
+
+variable "network_self_link" {
+  description = "The self link of the network in which the HTCondor central manager will be created."
+  type        = string
+  default     = null
+}
+
+variable "central_manager_service_account_email" {
+  description = "Service account for central manager (e-mail format)"
+  type        = string
+}
+
+variable "service_account_scopes" {
+  description = "Scopes by which to limit service account attached to central manager."
+  type        = set(string)
+  default = [
+    "https://www.googleapis.com/auth/cloud-platform",
+  ]
+}
+
+variable "network_storage" {
+  description = "An array of network attached storage mounts to be configured"
+  type = list(object({
+    server_ip             = string,
+    remote_mount          = string,
+    local_mount           = string,
+    fs_type               = string,
+    mount_options         = string,
+    client_install_runner = map(string)
+    mount_runner          = map(string)
+  }))
+  default = []
+}
+
+variable "disk_size_gb" {
+  description = "Boot disk size in GB"
+  type        = number
+  default     = null
+}
+
+variable "metadata" {
+  description = "Metadata to add to HTCondor central managers"
+  type        = map(string)
+  default     = {}
+}
+
+variable "enable_oslogin" {
+  description = "Enable or Disable OS Login with \"ENABLE\" or \"DISABLE\". Set to \"INHERIT\" to inherit project OS Login setting."
+  type        = string
+  default     = "ENABLE"
+  nullable    = false
+  validation {
+    condition     = contains(["ENABLE", "DISABLE", "INHERIT"], var.enable_oslogin)
+    error_message = "Allowed string values for var.enable_oslogin are \"ENABLE\", \"DISABLE\", or \"INHERIT\"."
+  }
+}
+
+variable "subnetwork_self_link" {
+  description = "The self link of the subnetwork in which the HTCondor central manager will be created."
+  type        = string
+  default     = null
+}
+
+variable "instance_image" {
+  description = "Custom VM image with HTCondor and Toolkit support installed."
```

Comment:
"and Toolkit support installed" means, more or less, the autoscaler virtual environment has been created. I'll change the text to read "Custom VM image with HTCondor installed using the htcondor-install module". We have the main example, but I'm wary about overloading variable descriptions when we already point to the example in the README.

We could do something like add a label to the image with Packer and enforce its existence using a terraform postcondition. There are organizations that enforce what labels are allowed to exist and this might break in those environments.

Reply:
I think this is fine

I am just realizing that you have to build an image. Is this a new restriction. Before (as vm-instance) would we have been able to install via startup-script without building an image.

Either way I am fine with this now that we have deploy command, it just means that the simplest form of the blueprint will always have 3 groups.

Reply:
I believe one could provide the install runners to the inputs. It is a bad experience, however, due to the significant setup time and makes autoscaling unimpressive. I think we should steer away from this as much as possible.

I think the 3-group pattern is going to be more common with time.